pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/DataType.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.vf;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import apex.jorje.semantic.symbol.type.BasicType;
+
+/**
+ * Represents all data types that can be referenced from a Visualforce page. This enum consolidates the data types
+ * available to CustomFields and Apex. It uses the naming convention of CustomFields.
+ *
+ * See https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_field_types.htm#meta_type_fieldtype
+ */
+public enum DataType {
```

Comment:
I renamed this from `IdentifierType`. This value can be stored on both `ASTIdentifier` and `ASTLiteral` nodes.